pkg/repository/git.go

```
@@ -113,12 +129,55 @@ type oktetoGitWorktree struct {
 	worktree *git.Worktree
 }
 
-func (ogr oktetoGitWorktree) Status() (gitStatusInterface, error) {
-	status, err := ogr.worktree.Status()
+func (ogr oktetoGitWorktree) GetRoot() string {
+	return ogr.worktree.Filesystem.Root()
+}
+
+func fixDubiousOwnershipConfig(path string) error {
+	c := exec.Command("git", "config", "--global", "--add", "safe.directory", path)
+	c.Dir = path
+	return c.Run()
+}
+
+func runGitStatusCommand(path string, fixAttempt int) (string, error) {
+	if fixAttempt > 0 {
+		return "", fmt.Errorf("failed to get status: too many attempts")
+	}
+	c := exec.Command("git", "status", "--porcelain", "-z")
+	c.Dir = path
+	output, err := c.Output()
+
+	if errors.Is(err, errors.New("detected dubious ownership in repository")) {
+		err = fixDubiousOwnershipConfig(path)
+		if err != nil {
+			return "", fmt.Errorf("failed to get status: cannot recover")
+		}
+		fixAttempt++
+		return runGitStatusCommand(path, fixAttempt)
+	}
+
+	return string(output), err
+}
+
+func (ogr oktetoGitWorktree) Status() (oktetoGitStatus, error) {
+	output, err := runGitStatusCommand(ogr.GetRoot(), 0)
 	if err != nil {
-		return nil, err
+		return oktetoGitStatus{status: git.Status{}}, fmt.Errorf("failed to get git status: %w", err)
 	}
-	return oktetoGitStatus{status: status}, nil
+
+	lines := strings.Split(string(output), "\000")
+	stat := make(map[string]*git.FileStatus, len(lines))
+	for _, line := range lines {
+		// line example values can be: "M modified-file.go", "?? new-file.go", etc
+		parts := strings.SplitN(strings.TrimLeft(line, " "), " ", 2)
+		if len(parts) == 2 {
+			stat[strings.Trim(parts[1], " ")] = &git.FileStatus{
+				Staging: git.StatusCode([]byte(parts[0])[0]),
+			}
+		}
```

should we return an error `if len(parts) != 2`